src/db.c

```diff
@@ -115,8 +115,8 @@ robj *lookupKey(serverDb *db, robj *key, int flags) {
         /* Update the access time for the ageing algorithm.
          * Don't do it if we have a saving child, as this will trigger
          * a copy on write madness. */
-        if (server.current_client && server.current_client->flag.no_touch && server.executing_client &&
-            server.executing_client->cmd->proc != touchCommand)
+        if (server.current_client && server.current_client->flag.no_touch &&
```

Comment:
reapply @PingXie suggestion
```suggestion
        if ((flags & LOOKUP_NOTOUCH) == 0 &&
            server.current_client && server.current_client->flag.no_touch &&
```